packages/next/src/server/dev/on-demand-entry-handler.ts

```diff
@@ -80,6 +80,33 @@ function convertDynamicParamTypeToSyntax(
   }
 }
 
+/**
+ * format: {compiler type}@{page type}@{page path}
+ * e.g. client@pages@/index
+ * e.g. server@app@app/page
+ *
+ * This guarantees the uniqueness for each page, to avoid conflicts between app/ and pages/
+ */
+
+export function getEntryKey(
+  compilerType: CompilerNameValues,
+  pageBundleType: 'app' | 'pages' | 'root',
+  page: string
+) {
+  return `${compilerType}@${pageBundleType}@${page}`
```

Comment:
Maybe it's better to just use `JSON.stringify([compilerType, pageBundleType, page])`?
It's easier to parse & handle.

Reply:
we need to extract the values from format in hot-reload, I believe this will be easier for now. We could change once entry key become complicated but seems enough for current case